pip_audit/_util.py

```diff
@@ -3,11 +3,36 @@
 """
 
 import sys
+import tempfile
+import os
 from typing import NoReturn  # pragma: no cover
 
 from packaging.version import Version
 
 
+class CustomNamedTemporaryFile:
+    """
+    Workaround for NamedTemporaryFile which also works on Windows. This 
+    version allows the created tempfile to be open and closed multiple times 
+    without losing permissions on the file on Windows file systems.
+    """
+    def __init__(self, mode='w+b'):
+        self._mode = mode
+
+    def __enter__(self):
+        # Generate a random temporary file
+        file_name = os.path.join(tempfile.gettempdir(), os.urandom(24).hex())
+        # Ensure the file is created
+        if not os.path.exists(file_name):
+            open(file_name, 'x').close()
+        # Open the file in the given mode
+        self._tempFile = open(file_name, self._mode)
+        return self._tempFile
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self._tempFile.close()
+        os.unlink(self._tempFile.name)
```

Comment:
A few thoughts:

* Rather than recreating most of the internals of `tempfile.NamedTemporaryFile`, can we re-use `tempfile.NamedTemporaryFile(delete=False)` for the file object here?
* Even if we don't use the "real" `NamedTemporaryFile` here, we should try to keep divergences from the "normal" tempfile behavior to a minimum. In particular, `os.urandom(24).hex()` looks like a reimplementation of `mkstemp`.